mutters-slots/src/main/java/com/rabidgremlin/mutters/slots/DateSlot.java

```diff
@@ -103,20 +89,64 @@ public SlotMatch match(String token, Context context)
 
         if (!dates.isEmpty() && percMatch > 0.75)
         {
-          DateTime theDateTime = new DateTime(dates.get(0),
-              DateTimeZone.forTimeZone(context.getTimeZone()));
-          return new SlotMatch(this, token, theDateTime.toLocalDate());
+          ZonedDateTime theDateTime = ZonedDateTime.ofInstant(dates.get(0).toInstant(), context.getTimeZone().toZoneId());
+          LocalDate localDate = theDateTime.toLocalDate();
+          return new SlotMatch(this, token, localDate);
         }
       }
     }
-
     return null;
   }
 
+  private LocalDate tryParse(String token, int currentYear, DateTimeFormatter fmt)
+  {
+    try
+    {
+      // workaround for migrating jodatime 'withDefault...' see: https://stackoverflow.com/a/49815584/6122976
+      // init the defaults
+      LocalDate defaults = LocalDate.of(currentYear, Month.JANUARY, 1);
+      // override default with parsed values
+      TemporalAccessor parsed = fmt.parse(token);
+      ChronoField[] fields = { ChronoField.YEAR, ChronoField.MONTH_OF_YEAR, ChronoField.DAY_OF_MONTH };
+      LocalDate result = defaults;
+      for (TemporalField field : fields) {
+        if (parsed.isSupported(field)) {
+          result = result.with(field, parsed.getLong(field));
+        }
+      }
+      return result;
+    }
+    catch (DateTimeParseException e)
+    {
+      return null;
+    }
+  }
+
+  private LocalDate tryParseWithPatternsInOrder(String token, Locale locale, int currentYear, String... fmtPatterns)
+  {
+    return Arrays.stream(fmtPatterns)
+        .distinct()
+        .map(fmtPattern -> tryParse(token, currentYear, DateTimeFormatter.ofPattern(fmtPattern).withLocale(locale)))
+        .filter(Objects::nonNull)
+        .findFirst()
+        .orElse(null);
+  }
+
+  private LocalDate tryParseOtherFormat(String token, Locale locale, int currentYear)
+  {
+    String[] shortFormPatternArray = { "dd/MM", "dd/M", "d/MM", "d/M",
```

Comment:
In Java 8 its recommended to do something like:
```
    Stream<String> shortFormPatterns = Stream.of(
        "[d][dd]/[M][MM]",
        "[d][dd]-[M][MM]",
        "[d][dd].[M][MM]",
        "[d][dd] [M][MM]",
        "[d][dd]/[M][MM]/[yyyy][yy]",
        "[d][dd]-[M][MM]-[yyyy][yy]",
        "[d][dd].[M][MM].[yyyy][yy]",
        "[d][dd] [M][MM] [yyyy][yy]");
```
```joda.time``` doesn't support the ```[]``` pattern.

Reply:
https://stackoverflow.com/questions/58160930/java8-equivalent-of-jodatime-datetimeformat-shortdate/58171902#58171902

Reply:
Can just use ```d[/][-][.][ ]M[/][-][.][ ][yyyy][yy]```